synapse/groups/groups_server.py

```diff
@@ -623,7 +623,14 @@ def invite_to_group(self, group_id, user_id, requester_user_id, content):
         )
 
         # TODO: Check if user knocked
-        # TODO: Check if user is already invited
+
+        invited_users = yield self.store.get_invited_users_in_group(group_id)
+        if user_id in invited_users:
+            raise SynapseError(400, "User already invited to group")
```

Comment:
Pleeeeeeease provide an error code to SynapseError. I really want to avoid future problems where `M_UNKNOWN` appears in the spec.

Reply:
@turt2live: any suggestions for what would be more appropriate?

Reply:
I thought we had an `M_BAD_REQUEST`, but it looks like not. Seems to be a theme in the spec to use M_UNKNOWN for 400 anyways.

Reply:
`M_BAD_STATE` seems good:
The state change requested cannot be performed, such as attempting to unban a user who is not banned.

`join` -> `invite` cannot be performed on a community